dbus_lynx_distributor/ftdi.py

```diff
@@ -1,186 +1,51 @@
-        self._write(data)
-
-    def set_i2c_stop(self):
-        data = bytes(
-            # Initial condition for the I2C Stop - Pull data low (Clock will already be low and is kept low)
-            [
-            0x80, # Command to set ADbus direction/data
-            0xFC, # put data and clock low
-            0xFB, # Set pins o/p except bit 2 (data_in)
-            ] * 4 + # Repeat commands to ensure the minimum period of the stop setup time is achieved
-            # Clock now goes high (open drain)
-            [
-            0x80, #Command to set ADbus direction/data
-            0xFD, #put data low, clock remains high
-            0xFB, #Set pins o/p except bit 2 (data_in)
-            ] * 4 + # Repeat commands to ensure the minimum period of the stop setup time is achieved
-            # Data now goes high too (both clock and data now high / open drain)
-            [
-            0x80, # Command to set ADbus direction/data
-            0xFF, # both clock and data now high
-            0xFB, # Set pins o/p except bit 2 (data_in)
-            ] * 4) # Repeat commands to ensure the minimum period of the stop hold time is achieved
-        self._write(data)
-
-    def read_byte_and_send_nak(self):
-        data = bytes([
-            0x20, # Command: clock data byte in on clk rising edge
-            0x00, # Length
-            0x00, # Length 0x0000 means clock ONE byte in
-            # Now clock out one bit (ACK/NAK). This bit has value '1' to send a NAK to the I2C Slave
-            0x13, # Command: clock data bits out on clk falling edge
-            0x00, # Length of 0x00 means clock out ONE bit
-            0xFF, # Command will send bit 7 of this byte (= ‘1’)
-            # Put I2C line back to idle (during transfer) state... Clock line low, Data line high
-            0x80, #Command to set ADbus direction/ data
-            0xFE, #Set the value of the pins
-            0xFB, #Set pins o/p except bit 2 (data_in)
-            # AD0 (SCL) is output driven low
-            # AD1 (DATA OUT) is output high (open drain)
-            # AD2 (DATA IN) is input (therefore the output value specified is ignored)
-            # AD3 to AD7 are inputs driven high (not used in this application)
-            # This command then tells the MPSSE to send any results gathered back immediately
-            0x87, # Send answer back immediate command
-            ])
-        self._write(data)
-
-        data_read = self._read(1)
-
+    
+    @property
+    def pid(self):
+        return f'0x{self._dev.idProduct:04x}'
+        
+    def init_i2c(self):
+        i2c = I2cController()
+        i2c.set_retry_count(1)
+        #Address format: ('ftdi://ftdi:pid:serial/1')
+        i2c.configure(f'ftdi://ftdi:{self.pid}:{self.serial_number}/1')
+        self.i2c = i2c
+       
+    def read_byte_and_send_nak(self, address: int):
+        port = self.i2c.get_port(address)
+        data_read = port.read(1)
```

Comment:
Here we also need to handle I2cNackError, or?

Reply:
agreed

Reply:
done